### PR TITLE
feat: Handle beam tasks asynchronously

### PR DESCRIPTION
**Pull request by Threated**
I have not tested this yet.

Doing this asynchronously with beam is also kind of difficult as we might get the same task again because the claiming is not done before we ask for the next one. So this PR will probably lead to the beam proxy being spammed by focus while we are claiming a task.

I want to implement a task streaming interface in beam which will prevent us from having this kind of issue but I have not gotten the time to do it.

### CODE REVIEW

**Comment by enola-dkfz** [comment]
tested on BBMRI-ERIC test env, thank you @torbrenner for adapting Lens

**Comment by Threated** [comment]
I moved a lot of functions away from main into other files as I created a new `task_processing.rs` file. There probably be even more stuff moved out of main but I dont want this PR to get much larger than it is already.

**Review by enola-dkfz — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -122,9 +131,9 @@ async fn main_loop() -> ExitCode {
             //TODO health check
         }
 
-        if let Err(e) = process_tasks(&mut obf_cache, &mut report_cache).await {
+        if let Err(e) = process_tasks(&mut task_queue, &mut seen_tasks).await {
             warn!("Encountered the following error, while processing tasks: {e}");
-            //failures += 1;
+            //failures += 1; //I believe this can be uncommented now
```
Comment (TKussel) [inline src/main.rs:136]: Please uncomment and test.

**Review by TKussel — APPROVED** [review]
Everything looks fine, however, I have not tested it in a lab environment.

File: src/main.rs (hunk 1 above)
Comment (enola-dkfz) [reply src/main.rs:136]: created issue, to be tested in another PR because it's not related to this

**Comment by enola-dkfz** [comment]
Thank you @PierreDelpy for testing in DKTK

**Pull request merged**